### PR TITLE
MULE-12671: WSC is not providing a way to pass transport headers

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: src/main/java/org/mule/extension/ws/internal/connection/SoapClientConnectionProvider.java [hunk 1]
```diff
@@ -73,31 +76,38 @@
...
   @Optional(defaultValue = "false")
   private boolean mtomEnabled;
 
   /**
+   * The transport configuration used to dispatch the SOAP messages.
+   */
+  @Placement(tab = "Transport")
+  @Parameter
+  @Optional
+  private CustomTransportConfiguration customTransportConfiguration;
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extension/ws/internal/connection/SoapClientConnectionProvider.java]: you add this, but where do you use it?

File: src/main/java/org/mule/extension/ws/internal/ConsumeOperation.java [hunk 2]
```diff
@@ -61,17 +64,19 @@
   @OutputResolver(output = ConsumeOutputResolver.class, attributes = ConsumeAttributesResolver.class)
   public Result<?, SoapAttributes> consume(@Connection SoapClient connection,
                                            @MetadataKeyId(OperationKeysResolver.class) String operation,
-                                           @ParameterGroup(name = "Message", showInDsl = true) SoapMessageBuilder message)
+                                           @ParameterGroup(name = "Message", showInDsl = true) SoapMessageBuilder message,
+                                           @Placement(tab = "Transport") @Optional @NullSafe Map<String, String> transportHeaders)
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extension/ws/internal/ConsumeOperation.java]: why isn't this part of the message?  Also, having a transport tab seems weird

File: src/main/java/org/mule/extension/ws/internal/connection/SoapClientConnectionProvider.java (hunk 1 above)
Comment (juandesi) [inline src/main/java/org/mule/extension/ws/internal/connection/SoapClientConnectionProvider.java]: I do use it in this class

File: src/main/java/org/mule/extension/ws/internal/ConsumeOperation.java (hunk 2 above)
Comment (juandesi) [inline src/main/java/org/mule/extension/ws/internal/ConsumeOperation.java]: because is not part of the soap message

Comment (alegmarra) [inline src/main/java/org/mule/extension/ws/internal/ConsumeOperation.java]: in SoapConnect we are not using a tab but a group `Transport Configuration` with these headers, and also declaring a DisplayName of `Headers`, which is also the displayname for message headers.

File: src/main/java/org/mule/extension/ws/internal/connection/SoapClientConnectionProvider.java [hunk 3]
```diff
@@ -53,18 +54,21 @@
   /**
    * The WSDL file URL remote or local.
    */
+  @Placement(order = 1)
   @Parameter
   private String wsdlLocation;
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/ws/internal/connection/SoapClientConnectionProvider.java]: can we provide an example for this?

File: src/main/java/org/mule/extension/ws/internal/TransportConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ */
+public class TransportConfiguration {
+
+  /**
+   * A group of transport headers that will be bounded with the transport request.
+   */
+  @Parameter
+  @Optional
+  @NullSafe
+  private Map<String, String> transportHeaders;
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/ws/internal/TransportConfiguration.java]: displayname

**Pull request merged**